### PR TITLE
Expand report <details> on print. Fixes #1240

### LINKED ISSUE #1240: Report: expand violation details on print
`<details>` not expanded in printed version, rendering it very less unusable since papers and PDF files are not that interactive, so you can't open them.

Should be expanded in print CSS.

### PR DESCRIPTION
**Pull request by ebidel**
R: all

The people have spoken. They want better printing!

### CODE REVIEW

**Comment by brendankenny** [comment]
FINALLY

File: lighthouse-core/report/scripts/lighthouse-report.js [hunk 1]
```diff
@@ -45,7 +45,32 @@ function sendJSONReport() {
   const popup = window.open(VIEWER_URL, '_blank');
 }
 
+/**
+ * Sets up listeners to expand audit `<details>` when the user prints the page.
```
Comment (brendankenny) [inline lighthouse-core/report/scripts/lighthouse-report.js]: this is kind of dumb, but maybe add a clause here that expanding `<details>` isn't possible via a print stylesheet? Whenever I think about this issue my first thought is always, "wait, why can't we just solve this in the print stylesheet?" and it probably will be again when I come across this function in the future :)

Comment (ebidel) [inline lighthouse-core/report/scripts/lighthouse-report.js]: done

**Comment by brendankenny** [comment]
viewer doesn't inherit this, right? Should it?

**Comment by ebidel** [comment]
> viewer doesn't inherit this, right? Should it?

It will :)

**Comment by brendankenny** [comment]
🖨 :1st_place_medal:

**Comment by paulirish** [comment]
Wow. The cross browser subtleties in this one are fascinating! Nice

**Pull request merged**